src/routes/group/chat/[id]/+page.svelte

```diff
@@ -160,18 +160,27 @@
 						<div class="messages">
 							<div class="messages-inner">
 								<!-- Chat bubbles -->
-								{#each messages as message}
+								{#each messages as message, i}
 									{#if message.type === 'user' && message.text.length > 0}
+										{@const len = messages.length}
+										{@const sameSender = messages[i].fromAddress === messages[i - 1]?.fromAddress}
+										{@const lastMessage =
+											i + 1 === messages.length ||
+											messages[i].fromAddress !== messages[i + 1]?.fromAddress ||
+											messages[(i + 1) % len].type !== 'user'}
```

Comment:
There is still modulo here which is incorrect. Try this instead:
```suggestion
											messages[i + 1]?.type !== 'user'}
```

Reply:
Sorry and thank you. I have a pretty bad headache today so I'm not really thinking straight. 😞

Reply:
Get better soon!